tinuous

```diff
@@ -393,9 +398,13 @@ class TravisJobLog(BuildLog):
             p,
         )
         r = self.client.get(f"/job/{self.job_id}/log.txt", stream=True)
-        with p.open("wb") as fp:
-            for chunk in r.iter_content(chunk_size=8192):
-                fp.write(chunk)
+        try:
+            with p.open("wb") as fp:
+                for chunk in r.iter_content(chunk_size=8192):
+                    fp.write(chunk)
+        except BaseException:
+            p.unlink(missing_ok=True)
+            raise
```

Comment:
seems like a reusable block of code, could be `stream_to_file(r, p)`

Reply:
Converted to function.